client/image.go

```diff
@@ -333,7 +333,11 @@ func (i *image) Unpack(ctx context.Context, snapshotterName string, opts ...Unpa
 	}
 
 	for _, layer := range layers {
-		unpacked, err = rootfs.ApplyLayerWithOpts(ctx, layer, chain, sn, a, config.SnapshotOpts, config.ApplyOpts)
+		curChain := append(chain, layer.Diff.Digest)
+		snapshotRefLabels := map[string]string{"containerd.io/snapshot.ref": identity.ChainID(curChain).String()}
+		snOpts := append(config.SnapshotOpts, snapshots.WithLabels(snapshotRefLabels))
```

Comment:
Looks like WithLabels from the caller is always overwritten.
BTW why not specify `SnapshotOpts` from the client side?

Reply:
hi @ktock, I'm not quite familiar with containerd internals, so may I ask what do you mean?
At least the `ctr` client doesn't have this, if `containerd.io/snapshot.ref` labels are guaranteed
by containerd behavior, which is documented in
https://github.com/containerd/containerd/blob/3a6e5fcb11a0fa826863077893eb4c68b82112a1/docs/remote-snapshotter.md?plain=1#L93
I guess this label should be tagged in containerd itself rather than each random client everywhere?

Reply:
IOWs, since it's a containerd-defined behavior, I think it'd be better to resolve it in containerd itself.
Otherwise the inconsistency due to random different clients will impact snapshotter behaviors too.